contrib/parseq-restli-client/src/main/java/com/linkedin/restli/client/config/RequestConfigTree.java

```diff
@@ -94,31 +95,35 @@ void add(RequestConfigElement element) {
     }
   }
 
+  /**
+   * This method recursively uses given resolver to resolve a config by given name taking into account
+   * syntax of sub-resource names. For example, for given name: Optional.of("foo:bar:baz") it will make
+   * the following resolver calls:
+   * - resolver(Optional.of("foo:bar:baz"))
+   * - resolver(Optional.of("foo:bar"))
+   * - resolver(Optional.of("foo"))
+   * - resolver(Optional.empty())
+   */
+  <X> Optional<X> resolveRecursivelyName(Optional<String> name, Function<Optional<String>, Optional<X>> resolver) {
```

Comment:
Why do we need generics X here? Is this always `Optional<ConfigValue<T>>`?

Reply:
For the logic of this method the only relevant aspect of return type is that it is Optional, thus it is defined as Optional<X>.
I realize that it might be easier to read in the context of surrounding code if it was Optional<ConfigValue<T>>, so I changed it to Optional<ConfigValue<T>>.

Reply:
Thanks.